temporal-sdk/src/main/java/io/temporal/internal/replay/ReplayWorkflowTaskHandler.java

```diff
@@ -247,27 +246,33 @@ private Result failureToWFTResult(
               .build();
       return new WorkflowTaskHandler.Result(workflowType, response, null, null, null, false);
     }
+
+    if (!shutdownFn.apply()) {
+      WorkflowExecution execution = workflowTask.getWorkflowExecution();
+      log.warn(
+          "Workflow task processing failure. startedEventId={}, WorkflowId={}, RunId={}. If seen continuously the workflow might be stuck.",
+          workflowTask.getStartedEventId(),
+          execution.getWorkflowId(),
+          execution.getRunId(),
+          e);
+    }
+
     // Only fail workflow task on the first attempt, subsequent failures of the same workflow task
     // should timeout. This is to avoid spin on the failed workflow task as the service doesn't
     // yet increase the retry interval.
     if (workflowTask.getAttempt() > 1) {
+      /*
+       *TODO we probably shouldn't swallow Error even if workflowTask.getAttempt() == 1.
+       * But leaving as it is for now, because a trivial change to rethrow
+       * will leave us without reporting Errors as WorkflowTaskFailure to the server,
+       * which we probably should at least attempt to do for visibility that the Error occurs.
+       */
       if (e instanceof Error) {
         throw (Error) e;
       }
       throw (Exception) e;
```

Comment:
Pardon my lack of JVM knowledge. Can you not just `throw e` instead of these 4 lines here (like you do in `catch (Throwable e)` blocks)? Why cast?

Reply:
Nevermind, catch clauses are special: https://stackoverflow.com/questions/23581611/why-is-it-legal-to-re-throw-a-throwable-in-certain-cases-without-declaring-it

Reply:
It's not because catch are specially really, but the problem that we catch all the Throwables to catch both Errors and Exceptions here and there theoretically can be a new subclass of Throwables that are not allowed to be thrown from a method that declares "throws Exception".